lightning/src/ln/channel.rs

```diff
@@ -4886,10 +4985,34 @@ pub(super) struct DualFundingChannelContext {
 	/// Note that the `our_funding_satoshis` field is equal to the total value of `our_funding_inputs`
 	/// minus any fees paid for our contributed weight. This means that change will never be generated
 	/// and the maximum value possible will go towards funding the channel.
+	///
+	/// Note that this field may be emptied once the interactive negotiation has been started.
 	#[allow(dead_code)] // TODO(dual_funding): Remove once contribution to V2 channels is enabled.
 	pub our_funding_inputs: Vec<(TxIn, TransactionU16LenLimited)>,
 }
 
+impl DualFundingChannelContext {
+	/// Obtain prev outputs for each supplied input and matching transaction.
+	/// Will error when a prev tx does not have an output for the specified vout.
+	/// Also checks for matching of transaction IDs.
+	fn txouts_from_input_prev_txs(inputs: &Vec<(TxIn, TransactionU16LenLimited)>) -> Result<Vec<&TxOut>, AbortReason> {
```

Comment:
Do we expect this will be used anywhere else? If not, I'd prefer to keep this closer to where it is called by either inlining or making a helper. Or maybe moving it to `interactive_tx.rs` near `calculate_change_output_value` for consistency. See other comment as well as this could be combined with `calculate_change_output_value` to avoid `Vec` allocations.

Reply:
This is likely will be needed in other places in the future.
Since the inputs are stored in the `DualFundingChannelContext`, I put it there as a helper.
I moved it to `interactivetxs.rs`, where it is internal now.

Reply:
I simplified it further, inlined this method.